README.md

```diff
@@ -26,22 +26,52 @@ A set of **clustergroupupgrade** example CRs can be found in the **samples** fol
+
+The states generally are represented by Conditions using appropriate reasons and types. The general workflow is:
+
+ClusterSelected -> BackupSucceeded -> PrecacheSpecValid -> PrecachingSucceeded -> Validated -> Progressing -> Succeeded
+
+Note that if Backup is not enabled then the BackupSuceeded condition will not be present, and similarly if precaching is not enabled then neither precaching condition will be present.
+
+A full table of the conditions and their appropriate reasons and types is:
+
+  | Type | Status| Reason| Message |
+  |------|-------|---------|--------|
+  ClustersSelected| True| Completed||
+  | |False | NotFound/NotPresent| `comment: mention the cluster list that is not part of the manangedCluster, must block the upgrade`|
+  Validated | True | Completed||
+  | | False | MissingManagedPolicy| `comment: must block the upgrade`|
+  | | False | InvalidPlatformImage| `comment: must block the upgrade`| 
+  `BackupSucceeded` | True | Completed | Backup is completed |
+  | | True | PartiallyDone | Failed to create backups for some clusters |
+  | | False | InProgress | Backup is in progress |
+  | | False | Failed | Backup failed for all the clusters |
+  `PrecacheSpecValid` | True | PrecacheSpecIsWellFormed/AsExpected | Pre-caching spec is valid and consistent|
+  | | False | NotAllManagedPoliciesExist| The ClusterGroupUpgrade CR has managed policies that are missing|
+  `PrecachingSucceeded` | True | Completed | Precaching is completed |
+  | | True | PartiallyDone | Failed to create preCaching for some clusters | 
+  | | False | InProgress | Precaching is in progress | 
+  | | False | Failed | PreCaching failed for all the clusters |
+  Progressing| True | InProgress||
+  | | False | Completed | |
+  | | False | NotEnabled| The ClusterGroupUpgrade CR is not enabled |
+  | | False | MissingBlockingCR | `comment: message must contain which blocking CRs are missing`|
+  | | False | IncompleteBlockingCR | `comment: blocked by other CR which is not completed, mention which CR is blocking`| 
+  Succeeded| True | UpgradeCompleted||
+  | | False | UpgradeTimedOut | |
+
+A few important ones to consider are:
+
+* **NotEnabled**
   * In this state, the **ClusterGroupUpgrade** CR has just been created and the *enable* field is set to *false*
   * The controller will build a remediation plan based on the *clusters* list and with *enable* fields like:
     * If *canaries* field is defined with a list of clusters, the first batch(es) of the remediation plan will contain those clusters
     * The number of remediation batches will be the length of *clusters* divided by *maxConcurrency*, each batch with a length of *maxConcurrency* containing the clusters following the *clusters* list ordering
```

Comment:
clusters list order is no longer followed

Reply:
Now I am thinking it can be useful for operators to specify the order if they want through the clusters list.

Reply:
Yes the entire cluster list is sorted before it is returned
I supposed we could just sort the label selectors part?

Reply:
Yes, that's what I am thinking. Don't worry about this sentence then. But maybe mention label selector two lines above?

Reply:
CNF-5558 reopened to track this